strict.js

```diff
@@ -0,0 +1,78 @@
+module.exports = {
+  root: true,
+  env: {
+    node: true,
+    es2022: true,
+  },
+  parser: "@typescript-eslint/parser",
+  plugins: [
+    "@typescript-eslint",
+    "import",
+    "jest",
+    "node",
+    "prettier",
+    "simple-import-sort",
+    "sonarjs",
+    "sort-destructure-keys",
+    "typescript-sort-keys",
+    "unicorn",
+  ],
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
+    "plugin:jest/recommended",
+    "plugin:node/recommended",
+    "plugin:prettier/recommended",
+    "plugin:sonarjs/recommended",
+    "plugin:typescript-sort-keys/recommended",
+    "plugin:unicorn/recommended",
+  ],
+  parserOptions: {
+    ecmaVersion: "latest",
+    project: "./tsconfig.json",
+    sourceType: "module",
+  },
+  rules: {
+    "import/default": "off",
+    "import/named": "off",
+    "import/namespace": "off",
+    "import/no-default-export": "error",
+    "import/no-extraneous-dependencies": [
+      "error",
+      { devDependencies: ["src/**/__tests__/**"] },
+    ],
+    "import/prefer-default-export": "off",
+    "jest/no-jest-import": "off",
+    "node/no-unpublished-import": [
+      "error",
+      {
+        allowModules: ["@jest/globals"],
+      },
+    ],
+    "node/no-unsupported-features/es-syntax": "off",
+    "node/no-missing-import": "off",
+    "simple-import-sort/imports": "error",
+    "simple-import-sort/exports": "error",
+    "sort-destructure-keys/sort-destructure-keys": "error",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
```

Comment:
It feels like some rules in the strict mode should be part of the default preset like this one

Reply:
It already is: https://github.com/open-turo/eslint-config-typescript/pull/118/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R81

Reply:
I see, I missed that. Then what we could do is have one preset extend from the other to avoid a lot of repetition and have a better understanding of the things one preset is adding

Reply:
Agreed. Although will have to explore how to create that hierarchy between JS files, as I'm assuming the usual `extends` won't be applicable here as a file reference (but maybe it works).

Related: https://github.com/open-turo/eslint-config-typescript/pull/118/files#r1025669400

Reply:
I think it should probably work fine but not completely sure, so we should give it a try

Reply:
The `extends` line does work internally, just with the path to the file.

It is not applicable to this PR, though, as the two presets are considered separate.

It does apply to the `eslint-config-react` PR, where an example can be found: https://github.com/open-turo/eslint-config-react/pull/85